course/functions.php

```diff
@@ -119,3 +111,65 @@ function course_theme_filter_single_lesson_template_for_sensei_learning_mode( $p
 	return $page_templates;
 }
 
+/**
+ * Add a body class with the variation.
+ *
+ * @param array $classes Body classes.
+ *
+ * @internal
+ *
+ * @return array Body classes.
+ */
+function add_body_class_for_variation( $classes ) {
+	$css_string    = wp_get_global_stylesheet( array( 'variables' ) );
+	$property_name = '--wp--custom--course-theme-variation';
+
+	// 1. "/": Delimiters that mark the start and end of the regex pattern.
+	// 2. "$property_name": This part of the pattern matches the specific property name, in our case, '--wp--custom--course-theme-variation', defined in Course theme's JSON files.
+	// 3. "\s*": Matches zero or more whitespace characters.
+	// 4. ":": Matches the colon you write to separate the CSS property name and property value.
+	// 5. "\s*": Matches zero or more whitespace characters after the colon.
+	// 6. "([^;]+)": This is a capturing group that matches one or more characters that are not a semicolon. It captures the value of the property.
+	// 7. "/": The closing delimiter of the regex pattern.
+	// Overall, this regex is designed to extract the value associated with a specific CSS property (defined in $property_name).
+	$pattern = "/$property_name\s*:\s*([^;]+)/";
+
+	$variation_name = 'default';
+
+	if ( preg_match( $pattern, $css_string, $matches ) ) {
+		// $matches[0] contains the full match.
+		// $matches[1] contains the CSS value for the specified property.
+		$css_value      = trim( $matches[1] );
+		$classes[]      = 'is-' . $css_value;
+		$variation_name = $css_value;
+	}
+
+	/**
+	 * Action to perform variation specific tasks.
+	 *
+	 * @hook course_theme_variation_loaded Fires after determining which theme variation is loaded.
+	 * @since $$next-version$$
+	 *
+	 * @param string $variation_name Name of the variation.
+	 */
+	do_action( 'course_theme_variation_loaded', $variation_name );
+
+	return $classes;
+}
+
+/**
+ * Enqueue the specific stylesheet for the variation.
+ *
+ * @param string $variation_name The current theme variation.
+ *
+ * @since Course $$next-version$$
```

Comment:
I think we should go ahead and add the proper version number here since we don't have a script we can run to replace `$$next-version$$` in this repo.

Reply:
Replaced it here with 1.3.5 https://github.com/Automattic/themes/pull/7432/commits/5c0a9b42710274d767638149e57baa72902ecdd6 ! Thanks! I'll go ahead and merge it as it is a very small change